src/main/java/build/buildfarm/worker/InputFetchStage.java

```diff
@@ -57,14 +57,11 @@ public void put(OperationContext operationContext) throws InterruptedException {
 
   @Override
   protected OperationContext tick(OperationContext operationContext) throws InterruptedException {
-    final String operationName = operationContext.operation.getName();
-    Poller poller = new Poller(worker.config.getOperationPollPeriod(), () -> {
-          boolean success = worker.instance.pollOperation(
-              operationName,
-              ExecuteOperationMetadata.Stage.QUEUED);
-          return success;
-        });
-    new Thread(poller).start();
+    Poller poller = workerContext.createPoller(
+        "InputFetchStage",
+        operationContext.operation.getName(),
+        ExecuteOperationMetadata.Stage.QUEUED,
+        () -> {});
```

Comment:
nit: () -> {} can be the default argument

Reply:
I meant an overload that only accepts three parameters.  https://stackoverflow.com/questions/997482/does-java-support-default-parameter-values